pkg/manager/task_clean.go

```diff
@@ -2,147 +2,367 @@ package manager
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"path/filepath"
-	"sync"
 
 	"github.com/stashapp/stash/pkg/image"
+	"github.com/stashapp/stash/pkg/job"
 	"github.com/stashapp/stash/pkg/logger"
 	"github.com/stashapp/stash/pkg/manager/config"
 	"github.com/stashapp/stash/pkg/models"
 	"github.com/stashapp/stash/pkg/plugin"
 	"github.com/stashapp/stash/pkg/utils"
 )
 
-type CleanTask struct {
-	ctx                 context.Context
-	TxnManager          models.TransactionManager
-	Scene               *models.Scene
-	Gallery             *models.Gallery
-	Image               *models.Image
-	fileNamingAlgorithm models.HashAlgorithm
+type cleanJob struct {
+	txnManager models.TransactionManager
+	input      models.CleanMetadataInput
+	scanSubs   *subscriptionManager
 }
 
-func (t *CleanTask) Start(wg *sync.WaitGroup, dryRun bool) {
-	defer wg.Done()
+func (j *cleanJob) Execute(ctx context.Context, progress *job.Progress) {
```

Comment:
A
```golang
        logger.Infof("Starting cleaning of tracked files")
        if j.input.DryRun {
                logger.Infof("Running in Dry Mode")
        }
```
should be added somewhere here ?  (it was removed due to the refactoring but not re-added)